src/components/ThemeEngine/StyleEngine.tsx

```diff
@@ -0,0 +1,76 @@
+import { CSS_PREFIX } from '../constants';
+import { kebabCase } from 'lodash';
+import React, { ReactNode, useMemo } from 'react';
+
+import { Root as PortalRoot } from '@radix-ui/react-portal';
+
+/**
+ * Validates a CSS color variable.
+ * @param value The value to validate
+ */
+const isColor = (value: string) => {
+  const s = new Option().style;
+  s.background = value;
+  return s.background !== '';
+};
+
+/**
+ * Adds children to document.head via Radix Portal.
+ * @param children The children to add to document.head
+ */
+const HeadPortal = ({ children }: { children: ReactNode }) => {
+  return (
+    <PortalRoot asChild={true} container={document.head}>
```

Comment:
And I don't think we need to pass `true` value for `boolean` props:

`<PortalRoot asChild={true}>` === `<PortalRoot asChild>`

Reply:
Yeah true - not sure which is preferrable. I have been being explicit with them in the past but I'm opening to dropping `={true}` in situations like this.

Reply:
Sweet, if y'all both prefer `asChild` I'll change it over.